Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -0,0 +1,410 @@
+%   PWI     - 3D image matrix, see explanation below
+%   PWI3D   - 4D image matrix, see explanation below
+%   PWI4D   - 4D image matrix, see explanation below
+%   x       - updated structure, with the following quantification parameters:
+%             fitting to the nomenclature explained below
+%   Control - same as PWI but then non-subtracted controls
+%   Control3D - same as PWI3D but then non-subtracted controls
+%   Control4D - same as PWI4D but then non-subtracted controls
+%
+%   x.Q.EchoTime_PWI4D
+%   x.Q.InitialPLD_PWI4D
+%   x.Q.LabelingDuration_PWI4D
+%
+%   x.Q.EchoTime_PWI3D
+%   x.Q.InitialPLD_PWI3D
+%   x.Q.LabelingDuration_PWI3D
+%
+%   x.Q.EchoTime_PWI
+%   x.Q.InitialPLD_PWI
+%   x.Q.LabelingDuration_PWI
+%
+% DESCRIPTION: This function performs subtraction and averaging all across the ExploreASL ASL modules, such as xASL_wrp_Realign,
+%              and xASL_wrp_RegisterASL to compute temporary PWI images, xASL_wrp_ResampleASL for creating QC images and preparing
+%              quantification, and xASL_wrp_Quantify if PWI4D is not used by the quantification algorithm (such as BASIL).
+%              It takes any PWI and control phase as input and will output any phase that it computed.
+%              The phases are: ASL4D->PWI4D    ->PWI3D    ->PWI
+%                         and: ASL4D->Control4D->Control3D->Control
+%
+% This function performs the following steps:
+% 1. Admin & checks
+% 2. Apply motion correction if this is needed
+%              
+% Nomenclature:
+% ASL4D = original timeseries
+% PWI4D = subtracted timeseries -> for quantification purposes (also applies to control4D)
+% PWI3D = averaged per TE—PLD—LabDur combination -> for QC purposes  (also applies to control3D)
+% PWI   = single perfusion-weighted volume -> for registration purposes  (also applies to control)
+% 
+% % we use the same nomenclature for control: control4D, control3D, control, as these are the same control-label pairs but without subtraction
+%
+% The quantification submodule xASL_wrp_Quantify only uses PWI4D from here, and recalculates PWI3D and PWI for quantification, potentially improving
+% quantification. So the user can, after QC, potentially remove some volumes from PWI4D, but should be aware that the information in the JSON sidecar
+% (TE, PLD, LabDur) should be adapted accordingly.
+%
+% EXAMPLE used by xASL_wrp_RegisterASL: PWI = xASL_im_ASLSubtractionAveraging(x, ASL_im);
+% EXAMPLE used by xASL_wrp_ResampleASL: [PWI, PWI3D, PWI4D, x] = xASL_im_ASLSubtractionAveraging(x, ASL4D, PWI4D, PWI3D);
+% __________________________________
+% Copyright (C) 2015-2023 ExploreASL
+
+
+%% ========================================================================================
+%% 1. Admin & checks
+% Input image volumes should be correct
+% Creation of the PWI3D or PWI4D image volumes can be skipped if they are provided as input
+
+if nargin<1 || isempty(x)
+    error('x input missing');
+end
+
+if nargin<2 || isempty(path_ASL4D)
```

Comment:
Wrong case.
You are actually using this option in
Functions/Quantification/xASL_quant_ASL.m:182:            PWI = xASL_im_ASLSubtractionAveraging(x, [], PWI4D);

Reply:
Yes, you wrote it in this way, but then add a parameter check that is not compatible with this way :D

I now changed the parameter checking that both input types for path_ASL4D - an empty string, and a matrix - are allowed.